kernel/src/snapshot.rs

```diff
@@ -142,22 +143,22 @@ impl Snapshot {
     }
 }
 
-#[derive(Debug, Deserialize, Serialize)]
+#[derive(Debug, Deserialize, Serialize, Schema)]
 #[serde(rename_all = "camelCase")]
 #[cfg_attr(feature = "developer-visibility", visibility::make(pub))]
 #[cfg_attr(not(feature = "developer-visibility"), visibility::make(pub(crate)))]
-struct CheckpointMetadata {
+struct LastCheckpointHint {
```

Comment:
hm, since the schema field can't be known at compile time I'm not sure we can `#[derive(Schema)]` for this..